ElectronClient/gui/KeymapConfigScreen.tsx

```diff
@@ -0,0 +1,229 @@
+import * as React from 'react';
+const CommandService = require('lib/services/CommandService').default;
+const { DOMtoElectronAccelerator } = require('lib/KeymapUtils.js');
+const { themeStyle } = require('lib/theme');
+const { _ } = require('lib/locale.js');
+// const { bridge } = require('electron').remote.require('./bridge');
+// const Setting = require('lib/models/Setting');
+
+interface KeymapItem {
+	label: string,
+	accelerator: string,
+	command: string
+	isEditing: boolean,
+}
+
+// Temporary values
+const commandNames = [
+	'textCopy',
+	'textCut',
+	'textPaste',
+];
+const getAccelerator = (commandName: string) => {
+	switch (commandName) {
+	case 'textCopy':
+		return 'Ctrl+C';
+	case 'textCut':
+		return 'Ctrl+X';
+	case 'textPaste':
+		return 'Ctrl+V';
+	default:
+		throw new Error('Invalid command');
+	}
+};
+
+const initialKeymap = commandNames.map(commandName => {
+	return {
+		label: CommandService.instance().label(commandName),
+		accelerator: getAccelerator(commandName),
+		command: commandName,
+		isEditing: false,
+	};
+});
+
+const KeymapConfigScreen = (props: { theme: Object }) => {
+	const [keymap, setKeymap] = React.useState([...initialKeymap]);
+	const [filter, setFilter] = React.useState('');
+
+	const theme = themeStyle(props.theme);
+	const containerStyle = {
+		...theme.containerStyle,
+		padding: 10,
+		overflow: 'auto',
+	};
+
+	const updateAccelerator = (commandName: string, accelerator: string) => {
+		const _keymap = [...keymap];
+
+		_keymap.find(item => item.command === commandName).accelerator = accelerator;
+		_keymap.find(item => item.command === commandName).isEditing = false;
+		setKeymap(_keymap);
+	};
+
+	const updateIsEditing = (commandName: string, isEditing: boolean) => {
+		const _keymap = [...keymap];
+
+		_keymap.forEach(item => item.isEditing = false);
+		_keymap.find(item => item.command === commandName).isEditing = isEditing;
+		setKeymap(_keymap);
+	};
+
+	return (
+		<div style={containerStyle}>
+			<TopActions
+				theme={props.theme}
+				filter={filter}
+				setFilter={setFilter}
+			/>
+			<KeymapTable
+				theme={props.theme}
+				keymap={keymap}
+				updateAccelerator={updateAccelerator}
+				updateIsEditing={updateIsEditing}
+			/>
+		</div>
+	);
+};
+
+const TopActions = (props: {
```

Comment:
Please put one component per file. Also please put everything under gui/KeymapConfig/

Reply:
I've made a folder and put the components in different files. However I included `TopActions` right into `KeymapConfigScreen` because it didn't seem necessary to have it's own component. The shortcut recorder however is defined in it's own file. I've also separated the styles from the component.